lib/puppet/provider/package/yum.rb

```diff
@@ -230,8 +230,10 @@ def install
 
       # FIXME: Should we raise an exception even if should == :latest
       # and yum updated us to a version other than @param_hash[:ensure] ?
-      vercmp_result = rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(is[:ensure]))
-      raise Puppet::Error, _("Failed to update to version %{should}, got version %{version} instead") % { should: should, version: is[:ensure] } if vercmp_result != 0
+      vercmp_result = is[:ensure].split(self.class::MULTIVERSION_SEPARATOR).any?{ |i|
+        0 == self.rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(i))
```

Comment:
Please use a `do..end` block style and a more meaningful variable name instead of `i`